### PR TITLE
Pulling back new changes from deployment

### PR DESCRIPTION
**Pull request by viktorcsimma**
This is to merge the improvements @horcsinbalint has made to the development branch.

@kdmnk what do you think? ~~I can review it, but maybe only on Sunday or after that.~~ Oh, the system doesn't let me do it; since it's technically my PR... but I can add comments, so it could be solved.

### CODE REVIEW

File: app/Console/Commands.php [hunk 1]
```diff
@@ -83,9 +83,10 @@ public static function latexToPdf($path, $outputDir)
         if (self::isDebugMode()) {
             $result = "ok";
         } else {
-            $command = "pdflatex " . "-interaction=nonstopmode -output-dir " . $outputDir . " " . $path . " 2>&1";
+            $command = "/usr/bin/pdflatex " . "-interaction=nonstopmode -output-dir " . $outputDir . " " . $path . " 2>&1";
```
Comment (kdmnk) [inline app/Console/Commands.php]: add the full path to the config then

File: app/Http/Middleware/TrustProxies.php [hunk 2]
```diff
@@ -12,7 +12,7 @@ class TrustProxies extends Middleware
      *
      * @var array|string
      */
-    protected $proxies;
+    protected $proxies = ['172.5.0.10'];
```
Comment (kdmnk) [inline app/Http/Middleware/TrustProxies.php]: What's this and why?

**Review by kdmnk — CHANGES_REQUESTED** [review]

File: app/Http/Middleware/TrustProxies.php (hunk 2 above)
Comment (horcsinbalint) [reply app/Http/Middleware/TrustProxies.php]: We are running mars behind a reverse proxy and setting the reverse proxy as a trusted proxy for Laravel is mandatory if we want to use HTTPS. (https://laravel.com/docs/9.x/requests#configuring-trusted-proxies) There are some other hacky ways of setting it up but I didn't like them.
I would like it to be configurable from the .env, and I will add this to the PR.

Comment (kdmnk) [reply app/Http/Middleware/TrustProxies.php]: Okay, and add a comment here. Will this affect local development?

**Comment by horcsinbalint** [comment]
Unfortunately I'm currently quite busy with my other tasks and other issues with the server. I'm planning on fixing the issues with this PR in about 3-4 weeks.

File: app/Http/Middleware/TrustProxies.php (hunk 2 above)
Comment (horcsinbalint) [reply app/Http/Middleware/TrustProxies.php]: I'm not 100% sure, and have not tested it, but I think it only comes in to effect if at least one of the headers in the request:
```
Request::HEADER_X_FORWARDED_FOR |
Request::HEADER_X_FORWARDED_HOST |
Request::HEADER_X_FORWARDED_PORT |
Request::HEADER_X_FORWARDED_PROTO |
Request::HEADER_X_FORWARDED_AWS_ELB
```
Which are mostly stuff that can come from a reverse proxy / load-balancer. But I think it should not have any effect unless you are connecting from IP `172.5.0.10` and the `APP_URL`'s protocol is https.

I will look into it today or tomorrow, but I think it should not have any effect for local development.

Comment (horcsinbalint) [reply app/Http/Middleware/TrustProxies.php]: I have tested it in my local environment and have not found any issues. So I think it should be fine

File: .env.example [hunk 3]
```diff
@@ -68,6 +68,8 @@ PREVENT_ACCESSING_MISSING_ATTRIBUTES=false
 PRINT_COST_ONESIDED=8
 PRINT_COST_TWOSIDED=12
 PRINTER_NAME=NemUjBela
+PRINTER_ADDITIONAL_ARGS=
+PRINTER_STAT_ADDITIONAL_ARGS=
```
Comment (BertalanD) [inline .env.example:72]: I'm not sure how much of the changes to printing will be necessary once the printer server is redesigned (eotvoscollegium/smaugreinstall#8). I'm planning on tackling that task on Thursday; could we delay this part until then?

File: app/Http/Middleware/TrustProxies.php (hunk 2 above)
Comment (BertalanD) [reply app/Http/Middleware/TrustProxies.php]: Could we make this configurable instead of hard-coding the IP in the source code?

Comment (horcsinbalint) [reply app/Http/Middleware/TrustProxies.php]: Yep, I would like to make it configurable, I just currently have a long lead-time

Comment (horcsinbalint) [reply app/Http/Middleware/TrustProxies.php]: Or if someone would fix it, I would happily give advice and / or check the code

**Comment by horcsinbalint** [comment]
I think I have fixed the issues that have been pointed out. @kdmnk @viktorcsimma Could you please check it?

Should fix #364 #366 #372 #373

**Review by kdmnk — APPROVED** [review]

**Pull request merged**